metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1017,6 +1041,24 @@ def get_upstream_tables(self, tables, datasource_name, browse_path, is_custom_sq
                 )
                 continue
 
+            # skip upstream tables if the source is postgres and the database is not whitelisted
+            if table.get(tableau_constant.CONNECTION_TYPE) == "postgres":
+                upstream_db = (
+                    table[tableau_constant.DATABASE][tableau_constant.NAME]
+                    if table.get(tableau_constant.DATABASE)
+                    and table[tableau_constant.DATABASE].get(tableau_constant.NAME)
+                    else ""
+                )
```

Comment:
Could also do something like this.
```suggestion
                upstream_db = table.get([tableau_constant.DATABASE], {}).get([tableau_constant.NAME], "")
```